src/v/kafka/server/replicated_partition.cc

```diff
@@ -485,8 +485,12 @@ replicated_partition::get_leader_epoch_last_offset_unbounded(
 ss::future<error_code> replicated_partition::prefix_truncate(
   model::offset kafka_truncation_offset,
   ss::lowres_clock::time_point deadline) {
+    if (kafka_truncation_offset == start_offset()) {
```

Comment:
`start_offset()` is not quite a trivial method. Consider caching it in a variable